### PR TITLE
Day9 Ex01 FeedFragment

### PR DESCRIPTION
**Pull request by at-hangtran**
Please review it for me! Thank you so much :)))

### CODE REVIEW

File: app/src/main/java/recyclerView/FriendActivity.java [hunk 1]
```diff
@@ -1,12 +1,12 @@
-package recyclerview;
+package recyclerView;
```
Comment (at-dev-support-01) [inline app/src/main/java/recyclerView/FriendActivity.java]: ERROR: Name 'recyclerView' must match pattern '^[a-z]+(\.[a-zA-Z_][a-zA-Z0-9_]*)*$'.

File: app/src/main/java/recyclerView/FriendAdapter.java [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package recyclerview;
+package recyclerView;
```
Comment (at-dev-support-01) [inline app/src/main/java/recyclerView/FriendAdapter.java]: ERROR: Name 'recyclerView' must match pattern '^[a-z]+(\.[a-zA-Z_][a-zA-Z0-9_]*)*$'.

File: app/src/main/java/recyclerView/FriendFragment.java [hunk 3]
```diff
@@ -1,4 +1,4 @@
-package recyclerview;
+package recyclerView;
```
Comment (at-dev-support-01) [inline app/src/main/java/recyclerView/FriendFragment.java]: ERROR: Name 'recyclerView' must match pattern '^[a-z]+(\.[a-zA-Z_][a-zA-Z0-9_]*)*$'.

File: app/src/main/java/recyclerView/User.java [hunk 4]
```diff
@@ -1,4 +1,4 @@
-package recyclerview;
+package recyclerView;
```
Comment (at-dev-support-01) [inline app/src/main/java/recyclerView/User.java]: ERROR: Name 'recyclerView' must match pattern '^[a-z]+(\.[a-zA-Z_][a-zA-Z0-9_]*)*$'.

File: app/src/main/java/recyclerView/FriendActivity.java (hunk 1 above)
Comment (at-dev-support-01) [inline app/src/main/java/recyclerView/FriendActivity.java]: WARNING: [PackageCase] Package name contains upper case characters
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/naming.html#PackageCase

File: app/src/main/java/recyclerView/FriendAdapter.java (hunk 2 above)
Comment (at-dev-support-01) [inline app/src/main/java/recyclerView/FriendAdapter.java]: WARNING: [PackageCase] Package name contains upper case characters
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/naming.html#PackageCase

File: app/src/main/java/recyclerView/FriendFragment.java (hunk 3 above)
Comment (at-dev-support-01) [inline app/src/main/java/recyclerView/FriendFragment.java]: WARNING: [PackageCase] Package name contains upper case characters
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/naming.html#PackageCase

File: app/src/main/java/recyclerView/User.java (hunk 4 above)
Comment (at-dev-support-01) [inline app/src/main/java/recyclerView/User.java]: WARNING: [PackageCase] Package name contains upper case characters
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/naming.html#PackageCase

File: app/build.gradle [hunk 5]
```diff
@@ -28,4 +34,7 @@ dependencies {
     compile 'com.android.support.constraint:constraint-layout:1.0.2'
     testCompile 'junit:junit:4.12'
     compile 'com.android.support:recyclerview-v7:25.3.1'
+    compile 'com.android.support:cardview-v7:22.0.0'
```
Comment (at-dev-support-01) [inline app/build.gradle]: ERROR: [GradleCompatible] This support library should not use a different version (22) than the `compileSdkVersion` (25)
 There are some combinations of libraries, or tools and libraries, that are incompatible, or can lead to bugs. One such incompatibility is compiling with a version of the Android support libraries that is not the latest version (or in particular, a version lower than your `targetSdkVersion`.)

Comment (at-dev-support-01) [inline app/build.gradle]: WARNING: [GradleDependency] A newer version of com.android.support:cardview-v7 than 22.0.0 is available: 25.3.1
 This detector looks for usages of libraries where the version you are using is not the current stable release. Using older versions is fine, and there are cases where you deliberately want to stick with an older version. However, you may simply not be aware that a more recent version is available, and that is what this lint check helps find.

File: app/build.gradle [hunk 6]
```diff
@@ -35,6 +35,7 @@ dependencies {
     testCompile 'junit:junit:4.12'
     compile 'com.android.support:recyclerview-v7:25.3.1'
     compile 'com.android.support:design:25.3.1'
+    compile 'com.android.support:cardview-v7:25.3.1'
```
Comment (at-vinhhuynh) [inline app/build.gradle]: add a beak line

File: app/src/main/AndroidManifest.xml [hunk 7]
```diff
@@ -18,5 +19,7 @@
         </activity>
         <activity android:name="recyclerview.FriendActivity" />
         <activity android:name=".ui.main.MainActivity" />
+        <activity android:name="demo.RecyclerViewActivity" />
     </application>
-</manifest>
+
+</manifest>
```
Comment (at-vinhhuynh) [inline app/src/main/AndroidManifest.xml]: please insert a break line

File: app/src/main/java/vn/asiantech/internship/feed/FeedAdapter.java [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+            mTvDescription = (TextView) itemView.findViewById(R.id.tvDescribe);
+            mViewPager = (ViewPager) itemView.findViewById(R.id.viewPager);
+        }
+    }
+
+    /**
+     * Used to register for viewpager.
+     */
+    private static class FeedPagerAdapter extends PagerAdapter {
+        private final List<Integer> images;
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/feed/FeedAdapter.java]: variable name `mImages`

File: app/src/main/java/vn/asiantech/internship/feed/FeedAdapter.java [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+        private final LayoutInflater inflater;
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/feed/FeedAdapter.java]: change variable name `mLayoutInflater `

File: app/src/main/res/layout/activity_feed.xml [hunk 10]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:orientation="vertical" android:layout_width="match_parent"
+    android:layout_height="match_parent">
+<FrameLayout
+    android:id="@+id/frContainer"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+</FrameLayout>
+</LinearLayout>
```
Comment (at-vinhhuynh) [inline app/src/main/res/layout/activity_feed.xml]: add a break line

File: app/src/main/java/vn/asiantech/internship/feed/FeedAdapter.java [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+/**
+ * Used to collect and display data on recyclerView.
+ *
+ * @author at-HangTran
+ * @version 1.0
+ * @since 2017-6-9
+ */
+class FeedAdapter extends RecyclerView.Adapter {
+
+    private List<Feed> feeds = new ArrayList<>();
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/feed/FeedAdapter.java]: variable name `mFeeds`

File: app/src/main/java/vn/asiantech/internship/feed/FeedAdapter.java [hunk 12]
```diff
@@ -24,12 +24,12 @@
  */
 class FeedAdapter extends RecyclerView.Adapter {
 
-    private List<Feed> feeds = new ArrayList<>();
+    private List<Feed> mFeeds = new ArrayList<>();
     private final Context mContext;
 
-    FeedAdapter(Context context, List<Feed> feeds) {
+    FeedAdapter(Context context, List<Feed> mFeeds) {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/feed/FeedAdapter.java]: change `mFeeds` on `constructor` to `feeds`

**Comment by at-vinhhuynh** [comment]
@at-hangtran 
Please resolve conflict

**Pull request merged**